src/components/Register.js

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import '../Register.css'
+import {register} from '../service/auth';
+import {Button, TextField} from "@material-ui/core";
+
+export class Register extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            login: '',
+            errorLoginText: '',
+            email: '',
+            emailPattern: /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
```

Comment:
If you don't change `emailPattern` in the state it might be move from the state outside (maybe even outside from the component)